app/src/main/java/com/yterletskyi/happyfriend/features/pin/ui/PinFragment.kt

```diff
@@ -3,31 +3,84 @@ package com.yterletskyi.happyfriend.features.pin.ui
 import android.os.Bundle
 import android.view.View
 import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
 import com.yterletskyi.happyfriend.R
 import com.yterletskyi.happyfriend.common.binding.BaseBindingFragment
 import com.yterletskyi.happyfriend.databinding.FragmentPinBinding
+import com.yterletskyi.happyfriend.features.pin.data.PinCode
+import com.yterletskyi.happyfriend.features.pin.data.PinCodeController
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
 
+@AndroidEntryPoint
 class PinFragment : BaseBindingFragment<FragmentPinBinding>(
     FragmentPinBinding::inflate
 ) {
+    @Inject
+    lateinit var pinCodeController: PinCodeController
 
     private val viewModel: PinViewModel by viewModels()
+    private val args: PinFragmentArgs by navArgs()
+    private var pin: String = ""
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         binding.pinKeyboardView.onButtonClicked = viewModel::input
 
+        with(binding.pinTitle) {
+            text = if(args.title != null) {
+                args.title
+            } else {
+                getString(R.string.pin_enter_title)
+            }
+        }
+
         viewModel.pinMaxLengthLiveData.observe(viewLifecycleOwner) { length ->
             binding.pinProgressView.steps = length
+
         }
 
         viewModel.pinProgressLiveData.observe(viewLifecycleOwner) { progress ->
             binding.pinProgressView.step = progress
+            authorize(progress)
         }
+    }
 
-        with(binding.pinTitle) {
-            text = getString(R.string.pin_enter_title)
+    private fun authorize(progress: Int) {
```

Comment:
let's please move this logic to ViewModel (or interactor). ViewModels should be responsible for view logic and interactor for business logic. We can discuss it in a call to make it more clear